CONTRIBUTING.md

```diff
@@ -82,3 +81,29 @@ Before a pull request can be merged:
 * The feature branch must be rebased so it contains the latest changes from the target branch
 * The CI has to pass successfully
 * Every comment has to be addressed and resolved
+
+## Testing
+
+Tests are written using the *[Ginkgo](https://onsi.github.io/ginkgo/)* framework. Here are some general advices when writing tests:
+
+* When the global `Describe` doesn't add enough context to the tests, use a nested `Describe` or a `Context` to specify what this test alludes to. Contexts should always start with _When_ (ie. "_When calling the function Foo_")
+* Start the descriptions of `It` blocks in lowercase and try to be as descriptive as possible
+* Avoid ignoring errors. In other words, make sure all of them are caught and tested
+* Files ending with `_suite_test.go` are meant to store the code that is common for tests in the same directory and that will be executed before them. Use this files only if your test setup is big enough to justify it (ie. the suite file has more than the test suite name and the teardown)
```

Comment:
Sentence two starts with
Use this files
I think `Use these files` (as opposed to `Use this file`) based on the way the rest of the line is written, but I think an argument can be made for either and would be cool with me